src/main/java/com/daml/extensions/testing/junit4/Sandbox.java

```diff
@@ -188,6 +177,14 @@ public DefaultLedgerAdapter getLedgerAdapter() {
     return sandboxManager.getLedgerAdapter();
   }
 
+  public Party getPartyIdForce(Party partyName) {
```

Comment:
I wonder why we called this "force"

Reply:
I have two "versions" of this functions getPartyIdForce and  getPartyId. The Force one allocate party on the ledger if it can't find it in the HashTable. The second one throws exception in this case.
The first ("force") needed to pass the tests. Some of the tests are trying to make actions on the ledger but no party is allocated. So I add this version of a function. 
But then I thought that maybe in our RLN project we will need more obvious one that throws exception. That's why I have to functions for this purpose.

Reply:
Can we call it `getPartyIdOrAllocate`?

Reply:
> Some of the tests are trying to make actions on the ledger but no party is allocated. So I add this version of a function.

Weird, how did they then work before? Is it not that you removed the "parties" section from `daml.yaml` and now those parties are not allocated?
I remember also having a section in the builder of the Sandbox that was used for party allocation. Even the README mentions it:
```
  private static Sandbox sandbox =
      Sandbox.builder()
          .dar(DAR_PATH)
          .moduleAndScript("Test", "testSetup")
          .parties(ALICE, BOB, CHARLIE)
          .build();
```

Reply:
Right! But this builder method `.parties(...)`  doesn't affect the ledger. 
`    public SandboxBuilder parties(String... parties) {
      this.parties = parties;
      return this;
    }`
It only in memory as I see. 
I'll try to bring back parties section to the daml.yaml, and then test without allocating. If it work I will get rid of "force" method. But As I remember I tried this way already.

Reply:
If `parties` doesn't affect the ledger, then that's a bug I would say.

Your getPartyIdOrAllocate (known as getPartyIdForce before) is a good idea IMO. I would like to have the parties allocated explicitly in each test instead of having them in daml.yaml, to be honest.
- in Cucumber, I do not know how
- in Java tests, one can use the builder + `parties(..)` (currently doesn't work as you noted)
Could you maybe call getPartyIdOrAllocate (known as getPartyIdForce before) for each element in `parties(..)`?